zap/src/main/java/org/zaproxy/zap/extension/pscan/scanner/StatsPassiveScanner.java

```diff
@@ -66,6 +69,28 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
         }
     }
 
+    private String getContentTypePostfix(String contentTypeValue) {
+        String[] ctvArray = contentTypeValue.split(";");
+        String contentType = "";
+        String charset = "";
+        if (ctvArray.length > 1) {
+            for (String segment : ctvArray) {
```

Comment:
Could just ignore the boundary, e.g. create a stream, (trim/remove empties,) filter out the boundary, and join.